### PR TITLE
[Android] Added draggable markers

### PR DESCRIPTION
**Pull request by iamanvesh**
#2450 

Ignored `MarkerView` as it is deprecated.

### CODE REVIEW

**Comment by iamanvesh** [comment]
@tobrun Could you please take a look at this PR?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java [hunk 1]
```diff
@@ -390,6 +397,57 @@ boolean onTap(PointF tapPoint) {
     return markerId != NO_ANNOTATION_ID && isClickHandledForMarker(markerId);
   }
 
+  //region Drag Event Handlers
+
+  private @Nullable Marker getMarkerBelowPoint(PointF point) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java]: This method could also be used by the `#onTap`, not to duplicate the code.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java [hunk 2]
```diff
@@ -447,6 +451,10 @@ public boolean onMoveBegin(MoveGestureDetector detector) {
     public boolean onMove(MoveGestureDetector detector, float distanceX, float distanceY) {
       // first move event is often delivered with no displacement
       if (distanceX != 0 || distanceY != 0) {
+        if (annotationManager.onMove(distanceX, distanceY)) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java]: You can skip `Projection#fromScreenLocation` conversion, and associated with it expensive JNI call, by passing coordinates returned by the `detector.getMoveObject(0)`.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java [hunk 3]
```diff
@@ -390,6 +397,57 @@ boolean onTap(PointF tapPoint) {
...
+    screenLocation.offset(-distanceX, -distanceY);
+    currentMarkerBeingDragged.setPosition(projection.fromScreenLocation(screenLocation));
+
+    if (onMarkerDragListener != null) {
+      onMarkerDragListener.onMarkerDrag(currentMarkerBeingDragged);
+    }
+    return true;
+  }
+
+  boolean onMoveEnd() {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java]: I feel like `onMoveEnd` logic should be moved to a separate method and invoked when there are more than 1 pointers placed on the screen to finish the drag.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/BaseMarkerOptions.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
   protected String snippet;
   protected String title;
   protected Icon icon;
+  protected boolean isDraggable = false;
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/BaseMarkerOptions.java]: No need for explicit initialization.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Marker.java [hunk 5]
```diff
@@ -40,6 +40,8 @@
   private int topOffsetPixels;
   private int rightOffsetPixels;
 
+  private boolean isDraggable = false;
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Marker.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java [hunk 6]
```diff
@@ -60,6 +61,8 @@
   private Polygons polygons;
   private Polylines polylines;
 
+  @Nullable private Marker currentMarkerBeingDragged = null;
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java (hunk 2 above)
Comment (iamanvesh) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java]: I haven't figured out how we can avoid `Projection#fromScreenLocation`. By using `detector.getMoveObject(0)` we can get the current screen coordinates which will avoid `Projection#toScreenLocation` call.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java [hunk 7]
```diff
@@ -385,9 +393,62 @@ boolean onTap(PointF tapPoint) {
...
+    }
+    return false;
+  }
+
+  boolean onMove(MoveDistancesObject movedObject, float distanceX, float distanceY) {
+    if (currentMarkerBeingDragged == null || !currentMarkerBeingDragged.isDraggable()) {
+      return false;
+    }
+    PointF screenLocation = new PointF(movedObject.getCurrentX(), movedObject.getCurrentY());
+    screenLocation.offset(-distanceX, -distanceY);
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java]: The current coordinates of the `MoveDistancesObject` reflect the most recent position, no need for offsetting them further.

**Comment by iamanvesh** [comment]
Are you referring to `OnMarkerDragListener` callbacks?

**Pull request closed without merging**